plugin/src/main/kotlin/com/ritense/plugin/service/PluginService.kt

```diff
@@ -106,6 +107,49 @@ class PluginService(
         return pluginConfiguration
     }
 
+    fun deployPluginConfigurations(deploymentDto: PluginAutoDeploymentDto){
+        val plugin: PluginConfiguration
+        val pluginDefinition = pluginDefinitionRepository.getById(deploymentDto.pluginDefinitionKey)
+        resolveProperties(deploymentDto.properties)
+        validateProperties(deploymentDto.properties!!, pluginDefinition)
+        plugin = if (deploymentDto.id != null){
```

Comment:
If there really is a use case for an nullable id field, this can be shortened by a bit:
```
plugin = pluginConfigurationRepository.save(PluginConfiguration(
                deploymentDto.id?.let { PluginConfigurationId.existingId(it) }?: PluginConfigurationId.newId(),
                deploymentDto.title,
                resolveProperties(deploymentDto.properties),
                pluginDefinition))
```

Reply:
fixed